### PR TITLE
feat(chat): add a 'Shared with me' section to the file manager modal (Issues #1283, #1982, #1987)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

- Added a 'Shared with me' section to the file manager modal.
- Implemented ability to add shared files to the App state.
- Fixed possibility to download published files from the 'Manage attachments' screen.
- Fixed usage of the published and shared files in the conversations.
- Fixed download files from publish and unpublish forms.

**Issues:**

- Issue #1283 
- Issue #1982 
- Issue #1987 

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1990.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Files/FileManagerModal.tsx [hunk 1]
```diff
@@ -175,6 +177,19 @@ export const FileManagerModal = ({
       searchQuery,
     ),
   );
+
+  const sharedWithMeRootFolders = useAppSelector((state) =>
```
Comment (IlyaBondar) [inline apps/chat/src/components/Files/FileManagerModal.tsx:170]: Please, add `sharedWithMeRootFolders` and `sharedWithMeRootFiles` into this fix (I suggest to merge fix into this branch and adapt for `shared with me`)
https://github.com/epam/ai-dial-chat/pull/2002/files

Comment (denys-kolomiitsev) [reply apps/chat/src/components/Files/FileManagerModal.tsx:170]: Thank you for the suggestion!
Branch merged. Made suggested changes.

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1990.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1990.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**